src/pages/MobileConnect/MobileConnect.tsx

```diff
@@ -61,14 +63,38 @@ const BodyText: React.FC<TextProps> = props => (
 )
 
 export const MobileConnect = () => {
-  const { create, importWallet, dispatch, getAdapter } = useWallet()
+  const { state, create, importWallet, dispatch, getAdapter } = useWallet()
+  const history = useHistory()
   const localWallet = useLocalWallet()
   const translate = useTranslate()
   const [wallets, setWallets] = useState<RevocableWallet[]>([])
   const [error, setError] = useState<string | null>(null)
   const [hideWallets, setHideWallets] = useState(false)
   const [isLoading, setIsLoading] = useState(true)
   const scaleFadeAnimation = `${scaleFade} 0.6s cubic-bezier(0.76, 0, 0.24, 1)`
+  const query = useQuery<{ returnUrl: string }>()
+
+  const hasWallet = Boolean(state.walletInfo?.deviceId)
+
+  useEffect(() => {
+    // This handles reloading an asset's account page on Native/KeepKey. Without this, routing will break.
+    // /:accountId/:assetId really is /:accountId/:chainId/:assetSubId e.g /accounts/eip155:1:0xmyPubKey/eip155:1/erc20:0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48
+    // The (/:chainId/:assetSubId) part is URI encoded as one entity in the regular app flow in <AssetAccountRow />, using generatePath()
+    // This applies a similar logic here, that works with history.push()
+    const match = matchPath<{ accountId?: string; chainId?: string; assetSubId?: string }>(
+      query.returnUrl,
+      {
+        path: '/accounts/:accountId/:chainId/:assetSubId',
+      },
+    )
+    const path = match
+      ? generatePath('/accounts/:accountId/:assetId', {
+          accountId: match?.params?.accountId ?? '',
+          assetId: `${match?.params?.chainId ?? ''}/${match?.params?.assetSubId ?? ''}`,
+        })
+      : query?.returnUrl
+    hasWallet && history.push(path ?? '/trade')
+  }, [history, hasWallet, query, state, dispatch])
```

Comment:
Effect moved hover from `<ConnecWallet />` - no changes here against develop, and mobile app was stuck on splash without it